client/src/main/java/com/tigrisdata/db/client/search/SearchFields.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2022 Tigris Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tigrisdata.db.client.search;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.tigrisdata.db.client.JSONSerializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+public class SearchFields implements JSONSerializable {
+
+  private final List<String> fields;
+  private static final SearchFields EMPTY = newBuilder().build();
+
+  private SearchFields(Builder builder) {
+    this.fields = Collections.unmodifiableList(builder.fields);
+  }
+
+  public List<String> get() {
```

Comment:
If this is supposed to be user facing API. please rename it to be more meaningful name.